src/style.rs

```diff
@@ -4,19 +4,81 @@ use crate::format::{BinaryBytes, DecimalBytes, FormattedDuration, HumanBytes, Hu
 use crate::state::ProgressState;
 use crate::utils::{expand_template, pad_str};
 use std::borrow::Cow;
+use std::collections::HashMap;
 
 #[cfg(feature = "improved_unicode")]
 use unicode_segmentation::UnicodeSegmentation;
 
+pub type Format = fn(&ProgressState) -> String;
```

Comment:
I don't think the `FormatMap` wrapper is useful here, since it's private to the crate anyway. Let's just rename `FormatMapInner` to `FormatMap`.

Reply:
Removed `FormatMapInner`.